src/crawlers/browser_crawler.js

```diff
@@ -360,8 +360,43 @@ export default class BrowserCrawler extends BasicCrawler {
      * @internal
      */
     async _handleRequestFunction(crawlingContext) {
-        const { id } = crawlingContext;
-        const page = await this.browserPool.newPage({ id });
+        const newPageOptions = {
+            id: crawlingContext.id,
+        };
+
+        if (this.proxyConfiguration && this.launchContext.useIncognitoPages) {
+            const pageOptions = {};
+
+            const { session } = crawlingContext;
+            const { url } = this.proxyConfiguration.newProxyInfo(session && session.id);
```

Comment:
I don't like that `newProxyInfo` does `new URL` and returns `url.href` instead of WHATWG URL :<

Reply:
Yeah, when we added it, not a lot of libs accepted WHATWG URL. Maybe we can change it in SDK V3.

Reply:
Ah, makes sense. Forgot that this code is quite old :)